cmake/ThirdParty.cmake

```diff
@@ -180,6 +180,31 @@ include_directories(SYSTEM
   ${CMAKE_CURRENT_BINARY_DIR}/3rdparty/cli11-prefix/include
 )
 
+#google test
+include(ExternalProject)
+
+ExternalProject_Add(
+    googletest
+    SOURCE_DIR "${CMAKE_BINARY_DIR}/googletest-src"
+    BINARY_DIR "${CMAKE_BINARY_DIR}/googletest-build"
+    GIT_REPOSITORY https://github.com/google/googletest.git
+    GIT_TAG release-1.11.0
+    GIT_SHALLOW ON
+    INSTALL_COMMAND ""
```

Comment:
Please take a look how other dependencies are added in ThirdParty.cmake.

We install each dependency to a prefix dir, then add prefix dir to include path and add specific libs using link_libraries().

We also use CMAKE_ARGS to pass all necessary cmake options to allow cross-compilation.

You also need to add gtest to ALL_DEPENDENCIES list.